src/plugins/data/common/index_patterns/index_patterns/types.ts

```diff
@@ -33,3 +33,16 @@ export interface TypeMeta {
   aggs?: Record<string, AggregationRestrictions>;
   [key: string]: any;
 }
+
+export interface IndexPatternSpec {
+  id: string;
+  type: string;
+  source: {
+    title: string;
+    timeFieldName: string;
+    sourceFilters: string;
+    fields: string;
```

Comment:
this should not be a string but an array of field specs (serializable respresentation if field)